reframe/frontend/executors/policies.py

```diff
@@ -49,7 +49,13 @@ def _print_perf(task):
     for key, info in perfvars.items():
         val, ref, lower, upper, unit, result = info
         name = key.split(':')[-1]
-        msg = f'P: {name}: {val} {unit} (r:{ref}, l:{lower}, u:{upper})'
+        
+        # Build reference info string only if all three reference value components are defined
+        if ref is not None and lower is not None and upper is not None:
```

Comment:
I would change the check here to


```suggestion
        if ref is not None:
```

Because otherwise it will masquerade cases where the reference is set, but not the thresholds. Also are you sure that checking the reference against `None` is correct. I think when it's missing we are setting it to 0:

https://github.com/reframe-hpc/reframe/blob/e3c4b3045df7ce3c086f9ce1143c46705dfbd19f/reframe/core/pipeline.py#L2440